src/Settings.cpp

```diff
@@ -19,6 +20,62 @@ static bool ends_with(const std::string& str, const std::string& with)
     return str.rfind(with) == str.size() - with.size();
 }
 
+void Settings::setUserConfigurationFile(const QString userConfigurationFileArg)
+{
+    userConfigurationFile = userConfigurationFileArg;
+}
+
+void Settings::setSettingsObject()
```

Comment:
For a clearer public interface, and for keeping it similar to the old one, I think this method should be made private. Then, it has to be called from every method using `settings`, like before, when we constructed a new object inside each method. All the code in the method should prevent double initialization checking that `settings` is equal to `nullptr` before initializing it.

Reply:
Another option would be to make a single initialize method that can optionally pass a user configuration file and it has to be called before any other use of the static class. If the user userConfigurationFileArg parameter is QString() (default) the standard format is used. The issue with this solution is that if some call to any other method is added before this initialization, the application will crash (it's not a major problem, since it will be easily spotted). This can already happen, by the way. The issue with the other option is that, if a call to a setValue or a getValue is added in the future, before the call to setUserConfigurationFile, that value will be written or read from the standard location instead of the specified configuration file. It might be better the crash, because it will be spotted, but I don't have a strong opinion about it.